hbase-server/src/main/java/org/apache/hadoop/hbase/replication/HBaseReplicationEndpoint.java

```diff
@@ -445,7 +418,7 @@ public PeerRegionServerListener(HBaseReplicationEndpoint endpoint) {
 
     @Override
     public synchronized void nodeChildrenChanged(String path) {
-      if (path.equals(regionServerListNode)) {
+      if (replicationEndpoint.fetchServersUseZk && path.equals(regionServerListNode)) {
```

Comment:
So in the new impl, it will always register zk listener but ignored the zk event if not use zk?

Reply:
Add comment in the class javadoc for this new impl?

Reply:
OK